Spigot-Server-Patches/0031-Fix-lag-from-explosions-processing-dead-entities.patch

```diff
@@ -5,27 +5,27 @@ Subject: [PATCH] Fix lag from explosions processing dead entities
 
 
 diff --git a/src/main/java/net/minecraft/server/Explosion.java b/src/main/java/net/minecraft/server/Explosion.java
-index bf910359de9365f51461ec1b80aaabdb24e26449..abc9eea66fed45db54e7b2a26ca8f7b05ca40e20 100644
+index bf910359de9365f51461ec1b80aaabdb24e26449..2288ca23ff274f3f83e7d021811d9b3efb20b534 100644
 --- a/src/main/java/net/minecraft/server/Explosion.java
 +++ b/src/main/java/net/minecraft/server/Explosion.java
 @@ -154,7 +154,7 @@ public class Explosion {
          int i1 = MathHelper.floor(this.posY + (double) f2 + 1.0D);
          int j1 = MathHelper.floor(this.posZ - (double) f2 - 1.0D);
          int k1 = MathHelper.floor(this.posZ + (double) f2 + 1.0D);
 -        List<Entity> list = this.world.getEntities(this.source, new AxisAlignedBB((double) i, (double) l, (double) j1, (double) j, (double) i1, (double) k1));
-+        List<Entity> list = this.world.getEntities(this.source, new AxisAlignedBB((double) i, (double) l, (double) j1, (double) j, (double) i1, (double) k1), (com.google.common.base.Predicate<Entity>) entity -> IEntitySelector.canAITarget().test(entity) && !entity.dead); // Paper - Fix lag from explosions processing dead entities
++        List<Entity> list = this.world.getEntities(this.source, new AxisAlignedBB((double) i, (double) l, (double) j1, (double) j, (double) i1, (double) k1), (com.google.common.base.Predicate<Entity>) entity -> IEntitySelector.notSpectator().test(entity) && !entity.dead); // Paper - Fix lag from explosions processing dead entities
```

Comment:
lets flip the dead check to be first and use isAlive()

so -> entity.isAlive() && !entity.isSpectator()

there is no reason to use the IEntitySelector here, so we can drop the obf helper as nothing else uses it once you clean up the other patches usage as i mentioned in the other review comment